public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -62,6 +65,20 @@ export class TablePanel extends Component<Props> {
     this.forceUpdate();
   };
 
+  onFilterAdded = (item: FilterItem) => {
+    const { key, value, operator } = item;
+    const panelModel = getDashboardSrv()
+      .getCurrent()
+      .getPanelById(this.props.id);
+    const datasource = panelModel?.datasource;
+
+    if (!datasource) {
+      return;
+    }
+
+    dispatch(applyFilterFromTable({ datasource, key, operator, value }));
```

Comment:
Not super happy with this, maybe we should introduce a prop on PanelProps that allows plugins to dispatch actions/events?

Reply:
question is what kind of actions/events? My thinking was to have an event bus on panel props, so panels can subscribe to and listen to events, ala https://github.com/grafana/grafana/pull/21010 
need to pick that PR up again. Got stuck on the event bus design and how to migrate or make it compatible with the panel events & app events (so we do not have two event systems).

Reply:
Ah yes, I remembered that one. Then I will just leave this as is for now.